src/commands/mod.rs

```diff
@@ -580,13 +580,23 @@ impl DaemonControl {
         }
     }
 
-    pub fn list_spend(&self) -> ListSpendResult {
+    pub fn list_spend(&self, txids: Option<Vec<bitcoin::Txid>>) -> ListSpendResult {
         let mut db_conn = self.db.connection();
-        let spend_txs = db_conn
-            .list_spend()
+        let spend_psbts = db_conn.list_spend();
+
+        let txids_set = txids
+            .as_ref()
+            .map(|list| list.iter().cloned().collect::<HashSet<_>>())
+            .unwrap_or_default();
+
+        let spend_txs: Vec<ListSpendEntry> = spend_psbts
             .into_iter()
+            .filter(|(psbt, _)| {
+                txids_set.is_empty() || txids_set.contains(&psbt.unsigned_tx.txid())
```

Comment:
I think this is incorrect. Or is it intended that passing an empty list would behave the same as not passing a list?

Reply:
right, its incorrect, should i error if an empty txid list is passed or return an empty Vec?

Reply:
I'm not sure. For instance in `listcoins` i think the behaviour is the same as what you implemented here, right @jp1ac4?

Reply:
Yep, for `listcoins`, an empty list means no filter, but in the GUI's `list_spend_transactions`, I was not consistent and used `None` for no filter:
https://github.com/wizardsardine/liana/blob/0e295acc8240e55b039d8e0fd56595a6d2ae13a0/gui/src/daemon/mod.rs#L96-L109

I guess we can still choose either option here depending on which makes more sense and then change any other commands to be consistent. I tend to think `None` for no filter makes sense and then an empty list doesn't require any extra logic as we just return whatever it contains, which could be nothing.

Reply:
What i find the most logical is (in JSON):
- `null`: no filter
- `[]`: a filter which rules out any spend tx

Now, the `[]` doesn't make any sense so i wouldn't mind if it meant "no filter". Just gotta be consistent across the board for when an empty list means "no filter" or "most extreme filter".

Reply:
I'm starting to think the best course of action here is to error in case of `[]` filter. If we find a use for it later on we can always change it. Better to go from Error -> No Error than to change its meaning.

Reply:
i've followed this way and added a test for this error case